plugins/modules/nxos_vrf_interfaces.py

```diff
@@ -0,0 +1,553 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2024 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The module file for nxos_vrf_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: nxos_vrf_interfaces
+short_description: Resource module to configure VRF interfaces.
+description: This module configures and manages the VRF configuration on interfaces on NX-OS platforms.
+version_added: 1.0.0
+author: Ruchi Pakhle (@Ruchip16)
+notes:
+  - Tested against Cisco NX-OS.
+  - This module works with connection C(network_cli).
+options:
+  config:
+    description: A list of interface VRF configurations.
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - Full name of the interface excluding any logical unit number,
+            i.e. Ethernet1/1.
+        type: str
+        required: true
+      vrf_name:
+        description:
+          - Name of the VRF to be configured on the interface.
+          - When configured, applies 'vrf member <vrf_name>' under the interface.
+        type: str
+  running_config:
+    description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the NX-OS
+        device by executing the command B(show running-config interface).
+      - The state I(parsed) reads the configuration from C(running_config)
+        option and transforms it into Ansible structured data as per the
+        resource module's argspec and the value is then returned in the
+        I(parsed) key within the result.
+    type: str
+  state:
+    description:
+      - The state the configuration should be left in
+    type: str
+    choices:
+      - merged
+      - replaced
+      - overridden
+      - deleted
+      - gathered
+      - rendered
+      - parsed
+    default: merged
+"""
+
+EXAMPLES = """
+# Using merged
+
+# Before state:
+# -------------
+#
+# nxos#show running-config interface
+# interface Ethernet1/1
+#  no switchport
+# interface Ethernet1/2
+#  description test
+#  no switchport
+#  no shutdown
+# interface Ethernet1/3
+# interface Ethernet1/4
+#  no switchport
+#  speed 1000
+#  no shutdown
+
+- name: Merge provided configuration with device configuration
+  cisco.nxos.nxos_vrf_interfaces:
+    config:
+      - name: Ethernet1/1
+      - name: Ethernet1/2
+        vrf_name: test
+      - name: Ethernet1/3
+      - name:Ethernet1/4
+    state: merged
+
+# Task Output:
+# ------------
+#
+# before:
+#   - name: "Ethernet1/1"
```

Comment:
so if no vrf is configured we still display interfaces?

Reply:
yess!